### PR TITLE
[risk=low][no ticket] Lint in PRs in parallel with unit tests

### PR DESCRIPTION
**Pull request by jmthibault79**
Benefits:
* Decreases the time of the actual test runs
* Linting errors are distinguishable from test errors
* Linting errors are visible more quickly

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [x] None of the above apply to this change.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -571,19 +574,34 @@ jobs:
...
+      - run:
+          name: Download Swagger CLI
+          working_directory: ~/workbench
+          command: ruby -r ./aou-utils/swagger.rb -e Workbench::Swagger.download_swagger_codegen_cli
+      - ui-cache
+      - run:
+          name: Yarn deps
+          working_directory: ~/workbench/ui
+          command: yarn deps test
       - run:
```
Comment (jmthibault79) [inline .circleci/config.yml:596]: this step often reveals lint problems, so I moved it as well

File: ui/src/app/pages/workspace/workspace-about.tsx [hunk 2]
```diff
@@ -2,15 +2,7 @@ import * as React from 'react';
 import * as fp from 'lodash/fp';
 import { faLock } from '@fortawesome/free-solid-svg-icons';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-
```
[the remark this answers is not in the document]
Comment (jmthibault79) [reply ui/src/app/pages/workspace/workspace-about.tsx:5]: no, this is an intentional linting error, to demonstrate the problem.  There's a similar one on the API side.

File: api/src/test/java/org/pmiops/workbench/actionaudit/ActionAuditServiceTest.java [hunk 3]
```diff
@@ -119,9 +119,7 @@ public void testSendWithEmptyCollectionDoesNotCallCloudLoggingApi() {
   private static final long AGENT_ID_2 = 102L;
   private static final String ACTION_ID = "b52a36f6-3e88-4a30-a57f-ae884838bfbf";
 
-  private static final ActionAuditEvent EVENT_1 =
-      ActionAuditEvent.builder()
-          .agentEmailMaybe("a@b.co")
+  private static final ActionAuditEvent EVENT_1 = ActionAuditEvent.builder()       .agentEmailMaybe("a@b.co")
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/actionaudit/ActionAuditServiceTest.java:122]: Causes api-lint to fail but not api-unit test

File: ui/src/app/pages/workspace/workspace-about.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (jmthibault79) [reply ui/src/app/pages/workspace/workspace-about.tsx]: Note: I removed this change to demonstrate that bad test formatting also causes a linting error.

**Review by NehaBroad — APPROVED** [review]

**Pull request merged**